openmdao/core/problem.py

```diff
@@ -791,6 +810,8 @@ def _calc_gradient_ln_solver(self, param_list, unknown_list, return_format, mode
         # Call arg > ln_solver option > auto-detect
         mode = self._mode(mode, param_list, unknown_list)
 
+        fwd = mode == 'fwd'
```

Comment:
it's pretty minor, but would be slightly faster if the 'if' is evaluated multiple times.